### PR TITLE
[v6r14] Matching jobs by hosts in a multi-VO environment

### PR DESCRIPTION
**Pull request by atsareg**
The hosts requesting jobs as GenericPilot must provide a VirtualOrganization parameter in the resource description, then they are allowed to match jobs.
Plus a bug fix in the resource description evaluation where Tags were sometimes overwritten

### CODE REVIEW

File: Resources/Computing/ComputingElement.py [hunk 1]
```diff
@@ -428,7 +428,8 @@ def getDescription( self ):
           tagList = []
           for i in range( 2, cores+1 ):
             tagList.append( '%dCore' % i )
-          ceDict['Tag'] = tagList  
+          ceDict.setdefault( 'Tag', [] )
```
Comment (chaen) [inline Resources/Computing/ComputingElement.py]: In one line ? 
ceDict.setdefault('Tag', []).extend(tagList)

Comment (atsareg) [inline Resources/Computing/ComputingElement.py]: I think we can announce a competition "Do your entire python program in just one line". There will be a lot of participants :)

Comment (chaen) [inline Resources/Computing/ComputingElement.py]: :D I agree that sometimes it is useless, but the reason I asked here is that I did not understand what you were trying to do in a first place. It's only after I looked at the whole file that I understood that there can already be a a value for tag, and that you create it only if it does not exist with this set defaut. So more than compressing to a maximum here (which I don't like either :-) ), I found it more comprehensible. But maybe it's only me :-)

File: WorkloadManagementSystem/Client/Matcher.py [hunk 2]
```diff
@@ -276,10 +276,14 @@ def _updatePilotJobMapping( self, resourceDict, jobID ):
...
     """ Check if we can get a job given the passed credentials
     """
-    # Check credentials if not generic pilot
     if Properties.GENERIC_PILOT in credDict[ 'properties' ]:
       # You can only match groups in the same VO
-      vo = Registry.getVOForGroup( credDict[ 'group' ] )
+      if credDict[ 'group' ] == "hosts":
+        # for the host case the VirtualOrganization parameter
+        # is mandatory in resourceDict
+        vo = resourceDict.get( 'VirtualOrganization', '' )
```
Comment (fstagni) [inline WorkloadManagementSystem/Client/Matcher.py]: So, vo here can be potentially empty. For LHCb case it really changes nothing, and I guess also for all single VO installations, but what about multi-VO installations?

File: Resources/Computing/ComputingElement.py (hunk 1 above)
Comment (fstagni) [inline Resources/Computing/ComputingElement.py]: I guess we are all fun of http://www.ioccc.org/

File: WorkloadManagementSystem/Client/Matcher.py (hunk 2 above)
Comment (atsareg) [inline WorkloadManagementSystem/Client/Matcher.py]: In multi-VO installations it will finally result in an exception and the jobs will not be matched

**Comment by fstagni** [comment]
Review OK

**Pull request merged**